pkg/controller/inferenceservice/resources/istio/virtualservice.go

```diff
@@ -251,8 +264,14 @@ func getTransformerStatusConfigurationSpec(endpointSpec *v1alpha2.EndpointSpec,
 		return nil, TransformerSpecMissing
 	}
 
+	if componentStatusMap == nil {
```

Comment:
Seeing all these checks for each compontent, I smell an opportunity to generalize. Workable?

Reply:
I think the way to do that is to have base component that allows us to treat this as a DAG. Out of scope of this humble change though.